src/storage/aggregator.cc

```diff
@@ -143,6 +161,108 @@ bool Aggregator::Update(const std::string& key, const std::string& row, const ui
     return true;
 }
 
+bool Aggregator::Init() {
+    if (GetStat() != AggrStat::kUnInit) {
+        PDLOG(INFO, "aggregator is normal or recovering");
+        return true;
+    }
+
+    status_.store(AggrStat::kRecovering, std::memory_order_relaxed);
+    if (!base_replicator_) {
+        status_.store(AggrStat::kUnInit, std::memory_order_relaxed);
```

Comment:
why not put `status_.store(AggrStat::kRecovering, std::memory_order_relaxed);` after the `if` statement. Then no need to restore the `kUnInit` stat.